src/master/master.cfg

```diff
@@ -205,18 +217,29 @@ for spec_id, spec in platform_manifest.iteritems():
                                         platform_id=spec_id,
                                         platform=spec,
                                         doStepIf=doStepIf,
+                                        sourceStamp={
+                                            # The `codebase` property must be
+                                            # set in order for this build step
+                                            # to function. It is currently
+                                            # undocumented:
+                                            # https://github.com/buildbot/buildbot/issues/4281
+                                            'codebase': u'',
+                                            'branch': None,
+                                            'repository': u'',
+                                            'revision': util.Property('announced_revision')
```

Comment:
Is this the magic bit that makes BuildBot check out the correct revision?

Reply:
It is. Took me over a day to figure it out :/